robot/impl/robot_reconfigure_weak_dependencies_test.go

```diff
@@ -278,11 +284,289 @@ func TestUpdateWeakDependents(t *testing.T) {
+	//    next level of reconfiguration.
+	//    The check will fail because there is nothing that depends on weak dependents, e.g., weak1, in the next level.
+	// 3) weak1 will configure.
+	// 4) At the end of reconfiguration, weak1 will reconfigure once (to increase weak1.reconfigCount to 1)
+	//    because of updateWeakDependents and `lRobot.lastWeakDependentsRound` will be updated to 5.
+	t.Log("Robot startup")
+	base1Name := base.Named("base1")
+	base2Name := base.Named("base2")
+	weakCfg := config.Config{
+		Components: []resource.Config{
+			{
+				Name:                weak1Name.Name,
+				API:                 weakAPI,
+				Model:               weakModel,
+				DependsOn:           []string{base1Name.String()},
+				ConvertedAttributes: &someTypeWithWeakAndStrongDepsConfig{},
+			},
+			{
+				Name:  base1Name.Name,
+				API:   base.API,
+				Model: fake.Model,
+			},
+			{
+				Name:  base2Name.Name,
+				API:   base.API,
+				Model: fake.Model,
+			},
+		},
+	}
+	robot := setupLocalRobot(t, context.Background(), &weakCfg, logger)
+
+	weakRes, err := robot.ResourceByName(weak1Name)
+	// The resource was found and all dependencies were properly resolved.
+	test.That(t, err, test.ShouldBeNil)
+	weak1, err := resource.AsType[*someTypeWithWeakAndStrongDeps](weakRes)
+	test.That(t, err, test.ShouldBeNil)
+	// Assert that the weak dependency was tracked.
+	test.That(t, weak1.resources, test.ShouldHaveLength, 2)
+	test.That(t, weak1.resources, test.ShouldContainKey, base1Name)
+	test.That(t, weak1.reconfigCount, test.ShouldEqual, 1)
+
+	lRobot := robot.(*localRobot)
+	test.That(t, lRobot.lastWeakDependentsRound.Load(), test.ShouldEqual, 4)
+
+	// Reconfigure base1
+	// The following scenario is expected:
+	// 1) The base1 will reconfigure (which will bump the logical clock).
+	// 2) The completeConfig loop will check to see if there is a need to updateWeakDependents by
+	//    checking the logical clock and also whether there are dependencies on weak dependents in
+	//    next level of reconfiguration.
+	//    The check will fail because there is nothing that depends on weak dependents, e.g., weak1, in the next level.
+	// 3) weak1 will not reconfigure, as base1 was neither newly built nor errored during reconfiguration.
+	// 4) At the end of reconfiguration, weak1 will reconfigure once (to increase weak1.reconfigCount to 2)
+	//    because of updateWeakDependents and `lRobot.lastWeakDependentsRound` will be updated to 5.
+	t.Log("Reconfigure base1")
+	weakCfg.Components[1].Attributes = rutils.AttributeMap{"version": 1}
+	robot.Reconfigure(context.Background(), &weakCfg)
+
+	weakRes, err = robot.ResourceByName(weak1Name)
+	test.That(t, err, test.ShouldBeNil)
+	weak1, err = resource.AsType[*someTypeWithWeakAndStrongDeps](weakRes)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, weak1.reconfigCount, test.ShouldEqual, 2)
+
+	test.That(t, lRobot.lastWeakDependentsRound.Load(), test.ShouldEqual, 5)
```

Comment:
So reconfiguring `base1` calls `SwapResource` and bumps the clock. But "weak reconfiguring" `weak1` does not result in a `SwapResource`?

Reply:
> yep, we don't call SwapResource when "weak reconfiguring."

Why do we not call `SwapResource` when weak reconfiguring?

Reply:
My understanding is that we don't want the logical clock to increase when weak reconfiguring.